### PR TITLE
Initial support for dockerized Spring Boot applications

### PR DESCRIPTION
**Pull request by bmuschko**
This functionality was covered in the blog post [Docker with Gradle: Dockerizing a Spring Boot application](https://bmuschko.com/blog/dockerized-spring-boot-app/). Given the popularity of the blog post I think it makes sense to roll it into the actual distribution as an opinionated plugin.

- Supports `jar` and `war`-based Spring Boot applications
- Uses the safer `CompileStatic` approach (making it easier to convert into an implementation that uses a statically compiled language)
- Works similar to the Docker Java application plugin
- Continues to use the convention mapping concept for uniformity reasons (should be [moved to the Provider concept](https://github.com/bmuschko/gradle-docker-plugin/issues/561) later)

### CODE REVIEW

File: src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy [hunk 1]
```diff
@@ -37,18 +37,9 @@ abstract class AbstractFunctionalTest extends Specification {
...
-
-        when:
-            BuildResult result = build('dockerVersion')
-
-        then:
-            result.output.contains('Retrieving Docker version.')
     }
 
     protected void setupBuildfile() {
-        if (buildFile) {
```
[the remark this answers is not in the document]
Comment (cdancy) [reply src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: Cleaning up as we go. I like it ;)

File: src/functTest/groovy/com/bmuschko/gradle/docker/DockerSpringBootApplicationPluginFunctionalTest.groovy [hunk 2]
```diff
@@ -0,0 +1,302 @@
+package com.bmuschko.gradle.docker
+
+import spock.lang.Requires
+import spock.lang.Unroll
+
+class DockerSpringBootApplicationPluginFunctionalTest extends AbstractFunctionalTest {
+    private static final String PROJECT_NAME = 'spring-boot'
+    private static final String CUSTOM_BASE_IMAGE = 'openjdk:7-alpine'
```
Comment (cdancy) [inline src/functTest/groovy/com/bmuschko/gradle/docker/DockerSpringBootApplicationPluginFunctionalTest.groovy]: Might consider changing to `openjdk:8-alpine` like what you did with the `DockerJavaApplicationPluginFunctionalTest` if only to keep things the same across the board. Might make sense to create a single property used by all tests.

**Comment by cdancy** [comment]
@orzeh when you get a chance please give things a look over to see if anything sticks out or doesn't make sense as we will have to maintain this going forward.

File: src/functTest/groovy/com/bmuschko/gradle/docker/DockerSpringBootApplicationPluginFunctionalTest.groovy (hunk 2 above)
Comment (bmuschko) [reply src/functTest/groovy/com/bmuschko/gradle/docker/DockerSpringBootApplicationPluginFunctionalTest.groovy]: done

**Comment by orzeh** [comment]
LGTM :+1:

**Comment by bmuschko** [comment]
Thanks for reviewing!

**Pull request merged**